src/Factory/ProductListViewFactory.php

```diff
@@ -101,14 +96,19 @@ private function getImageViews(Collection $images)
 
     /**
      * @param Collection|TaxonDocument[] $taxons
+     * @param TaxonDocument|null $mainTaxonDocument
      *
-     * @return TaxonView[]
+     * @return array
      */
-    private function getTaxonViews(Collection $taxons)
+    private function getTaxonCodes(Collection $taxons, TaxonDocument $mainTaxonDocument = null)
```

Comment:
`TaxonDocument $mainTaxonDocument = null` -> `?TaxonDocument $mainTaxonDocument` - let's make this parameter obligatory, should be less confusing